server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -134,14 +139,15 @@ public void heartbeat() {
 
         for (PluginDescriptor descriptor : elasticAgentPluginRegistry.getPlugins()) {
             List<ClusterProfile> clusterProfiles = clusterProfilesService.getPluginProfiles().findByPluginId(descriptor.id());
+            resolveSecrets(clusterProfiles);
```

Comment:
If secrets resolution for one of the cluster profile fails, the plugin would not get a ping for any of the clusters. 

How do we notify users in cases of secret resolution failures?

Reply:
If secret resolution fails for any cluster profiles, the server ping call will do for the remaining of the clusters. Also, we will not be able to notify the users about the agents for which the plugin has been deleted.
While the error gets logged in server logs, do you think we should catch them nd display a server health message?